src/__tests__/getLocalFileName.spec.ts

```diff
@@ -0,0 +1,22 @@
+import path from 'path';
+
+import { getLocalFileName } from '../getLocalFileName';
+
+describe('test getting local file name', () => {
+  const url = 'https://filesamples.com/samples/document/txt/sample3.txt'
+  const { name, ext } = path.parse(url);
+  const result = getLocalFileName(url);
+  const hashLength = 96;
+
+  test('local file name starts with correct name', () => {
+    expect(result.slice(0, name.length + 1)).toBe('sample3-');
```

Comment:
nit: String.[startsWith|endsWith] consider to try it

Reply:
done